src/logic/safe/hooks/useOwnerSafes.ts

```diff
@@ -8,39 +8,54 @@ import useCachedState from 'src/utils/storage/useCachedState'
 
 type OwnedSafesCache = Record<string, Record<string, string[]>>
 
+type UseOwnerSafesType = {
+  isLoading: boolean
+  ownerSafes: Record<string, string[]>
+}
+
 const storageKey = 'ownedSafes'
 
-const useOwnerSafes = (): Record<string, string[]> => {
+const useOwnerSafes = (): UseOwnerSafesType => {
   const connectedWalletAddress = useSelector(userAccountSelector)
   const chainId = useSelector(currentChainId)
   const [cache = {}, setCache] = useCachedState<OwnedSafesCache>(storageKey)
   const ownerSafes = cache[connectedWalletAddress] || {}
+  const [isLoading, setIsLoading] = useState<boolean>(true)
 
   useEffect(() => {
     if (!connectedWalletAddress) {
       return
     }
+    let isCurrent = true
 
     const load = async () => {
       try {
         const safes = await fetchSafesByOwner(connectedWalletAddress)
-
-        setCache((prev = {}) => ({
-          ...prev,
-          [connectedWalletAddress]: {
-            ...(prev[connectedWalletAddress] || {}),
-            [chainId]: safes,
-          },
-        }))
+        if (isCurrent) {
+          setCache((prev = {}) => ({
+            ...prev,
+            [connectedWalletAddress]: {
+              ...(prev[connectedWalletAddress] || {}),
+              [chainId]: safes,
+            },
+          }))
+          setIsLoading(false)
+        }
       } catch (err) {
-        logError(Errors._610, err.message)
+        if (isCurrent) {
+          logError(Errors._610, err.message)
+          setIsLoading(false)
```

Comment:
Don't call `setIsLoading()` twice with the same value. You could either move it to the `finally` block or simply outside the `try/catch`.